layer/flavors/custom.py

```diff
@@ -0,0 +1,51 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any
+
+import cloudpickle  # type: ignore
+from layerapi.api.entity.model_version_pb2 import ModelVersion
+
+from layer.types import ModelArtifact
+
+from .base import ModelFlavor
+
+
+class CustomModel:
+    """
+    A generic model that evaluates inputs and produces outputs.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initializes the custom  model
+        """
+
+    @abstractmethod
+    def predict(self, model_input: Any) -> Any:
```

Comment:
To provide support for the new prediction api for such custom models, this function should have more specific signature. E.g it accepts a pandas dataframe and returns a pandas dataframe

Reply:
Fixed